skyrl-train/examples/mini_swe_agent/mini_swe_utils.py

```diff
@@ -0,0 +1,95 @@
+from typing import TypedDict, Optional
+import traceback
+import uuid
+
+from typing import Dict, Any
+from loguru import logger
+
+from jinja2 import Template
+
+from minisweagent.environments import Environment, get_environment
+
+
+class MiniSWEEvaluationResult(TypedDict):
+    instance_id: str
+    resolved: bool
+    eval_error: Optional[str]
+
+
+def get_sb_environment(config: dict, instance: dict, data_source: str) -> Environment:
+    env_config = config.setdefault("environment", {})
+    env_config["environment_class"] = env_config.get("environment_class", "docker")
+    image_name = get_docker_image_name(instance, data_source)
+    if env_config["environment_class"] == "docker":
+        env_config["image"] = image_name
+    elif env_config["environment_class"] == "singularity":
+        env_config["image"] = f"docker://{image_name}"
+    env = get_environment(env_config)
+    if startup_command := config.get("run", {}).get("env_startup_command"):
+        startup_command = Template(startup_command).render(**instance)
+        out = env.execute(startup_command)
+        if out["returncode"] != 0:
+            raise RuntimeError(f"Error executing startup command: {out}")
+    return env
+
+
+def get_docker_image_name(instance: dict, data_source: str) -> str:
+    """Get the image name for a SWEBench/SWE-Gym instance."""
+    image_name = instance.get("image_name", None)
+    if image_name is None:
+        iid = instance["instance_id"]
+        if "swe-gym" in data_source.lower():
+            id_docker_compatible = iid.replace("__", "_s_")  # to comply with docker image naming convention
+            image_name = f"docker.io/xingyaoww/sweb.eval.x86_64.{id_docker_compatible}:latest".lower()
+        elif "swe-bench" in data_source.lower():
+            # Docker doesn't allow double underscore, so we replace them with a magic token
+            id_docker_compatible = iid.replace("__", "_1776_")
+            image_name = f"docker.io/swebench/sweb.eval.x86_64.{id_docker_compatible}:latest".lower()
+        else:
+            raise NotImplementedError(f"Data source: {data_source} is not supported")
+    return image_name
+
+
+def evaluate_trajectory(
+    instance: Dict[str, Any], model_patch: str, sweagent_config: dict, data_source: str
+) -> MiniSWEEvaluationResult:
+    instance_id = instance["instance_id"]
+
+    ret = MiniSWEEvaluationResult(instance_id=instance_id, resolved=False, eval_error=None)
```

Comment:
given `instance_id` is only used once, let's inline it in this call like
```
MiniSWEEvaluationResult(instance_id=instance["instance_id"], resolved=False, eval_error=None)
```
and remove the variable.